### PR TITLE
Update table docs.

### PR DESCRIPTION
**Pull request by markstory**
The buildRules() method is slightly different than other hooks as
it doesn't use an event on the Table class.

Refs #14250

### CODE REVIEW

File: src/ORM/Table.php [hunk 1]
```diff
@@ -96,8 +96,10 @@
  * - `buildValidator(EventInterface $event, Validator $validator, string $name)`
  *   Allows listeners to modify validation rules for the provided named validator.
  *
- * - `buildRules(EventInterface $event, RulesChecker $rules)`
- *   Allows listeners to modify the rules checker by adding more rules.
+ * - `buildRules(RulesChecker $rules)`
+ *   Allows table classes to modify the rules checker by adding more rules. The
+ *   `Model.buildRules` event will also include an `$event` parameter before the
```
Comment (othercorey) [inline src/ORM/Table.php]: This whole docblock section is confusing. Why are we mixing callbacks and events together?

The comment about `Model.buildRules` is confusing as it doesn't explain the relationship between Table::buildRules and Model.buildRules.

File: src/ORM/Table.php [hunk 2]
```diff
@@ -96,8 +96,10 @@
  * - `buildValidator(EventInterface $event, Validator $validator, string $name)`
```
Comment (othercorey) [inline src/ORM/Table.php]: Why do we document these without the `Model.` prefix?

Comment (markstory) [reply src/ORM/Table.php]: These are the table methods. Your comment about conflating the methods and events is spot on. I will split up the two sections.

File: src/ORM/Table.php [hunk 3]
```diff
@@ -81,51 +81,59 @@
  * You can use Table::updateAll() and Table::deleteAll() to do bulk updates/deletes.
  * You should be aware that events will *not* be fired for bulk updates/deletes.
  *
- * ### Callbacks/events
+ * ### Events
  *
- * Table objects provide a few callbacks/events you can hook into to augment/replace
- * find operations. Each event uses the standard event subsystem in CakePHP
+ * Table objects emit several events during as life-cycle hooks during find, delete and save
```
Comment (garas) [inline src/ORM/Table.php]: ... during as life-cycle hooks during ...

**Pull request merged**